### PR TITLE
Pr17 modified

### PR DESCRIPTION
**Pull request by makoto**
Modification of https://github.com/ensdomains/address-encoder/pull/17/

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -51,8 +51,8 @@ function makeBase58CheckDecoder(p2pkhVersions: number[], p2shVersions: number[])
 
 const base58Chain = (name: string, coinType: number, p2pkhVersions: number[], p2shVersions: number[]) => ({
```
Comment (Arachnid) [inline src/index.ts:52]: If we're renaming the functions we should rename this too.

File: src/index.ts [hunk 2]
```diff
@@ -250,6 +250,46 @@ function strEncoder(data: Buffer): string {
   return encodeEd25519PublicKey('ed25519PublicKey', data)
 }
 
+function tezosAddressEncoder(data: Buffer): string {
+  if (data.length !== 22 && data.length !== 21) { throw Error('Unrecognised address format'); }
+
+  let prefix: Buffer;
+  switch (data.readUInt8(0)) {
+    case 0x00:
```
Comment (Arachnid) [inline src/index.ts:258]: Is this version byte canonical, or something we're adding? If it's something we're adding, are the two address formats incompatible, or is there a tz1/tz2/tz3 address for each KT1 address?

File: src/index.ts [hunk 3]
```diff
@@ -250,6 +250,46 @@ function strEncoder(data: Buffer): string {
   return encodeEd25519PublicKey('ed25519PublicKey', data)
 }
 
+function tezosAddressEncoder(data: Buffer): string {
```
Comment (Arachnid) [inline src/index.ts:253]: It'd be a good idea to link to some documentation on these formats along with the function, so people can verify it's doing the right thing.

File: package.json [hunk 4]
```diff
@@ -41,6 +41,7 @@
   },
   "dependencies": {
     "bech32": "^1.1.3",
-    "crypto-addr-codec": "^0.1.7"
+    "crypto-addr-codec": "^0.1.7",
+    "eztz-lib": "^0.1.2"
```
Comment (Arachnid) [inline package.json:45]: I don't see this being used anywhere.

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (makoto) [reply src/index.ts:258]: it was implemented by the original contributor so I am not certain but it looks the version byte is mentioned here

https://tezos.stackexchange.com/questions/183/base58-encoding-decoding-of-addresses-in-micheline

File: src/index.ts (hunk 3 above)
Comment (makoto) [reply src/index.ts:253]: The only relevant reference I could find was from the stack exchange above. 

https://tezos.gitlab.io/api/p2p.html?highlight=contract_id#contract-id-22-bytes-8-bit-tag

I will add them as comments.

**Comment by makoto** [comment]
Please review again @Arachnid

**Comment by krivenkoviktor1954** [comment]
You took my code but didn't pay me

**Pull request merged**